### ISSUE TITLE
[no-use-before-define] False positives after upgrading from 4.0.1 to 4.1.0

### ISSUE THREAD

**Issue by EvHaus** [typescript-eslint/typescript-eslint#2520]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-use-before-define": "error"
  },
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  }
}
```

Using `@typescript-eslint/parser` version 4.0.1 the following code works just fine:

```TS
import React from 'react';

const Component = () => <div />;

export default Components;
```

However, after upgrading the parser to 4.1.0, I'm getting this error:

```
| 1:8  error  'React' was used before it was defined  no-use-before-define
```

```
# tsconfig.json
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "module": "commonjs",
    "lib": ["es2019", "dom"],
    "target": "esnext",
    "jsx": "react",
    "baseUrl": "./",
    "strict": true,
    "paths": {
      "*": [
        "src/*"
        "node_modules/*"
      ]
    },
    "resolveJsonModule": true
  }
}
```

**Expected Result**

There should be no error on this file.

**Actual Result**

ESLint error on:

```
| 1:8  error  'React' was used before it was defined  no-use-before-define
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.1.0` |
| `TypeScript`                | `4.0.2` |
| `ESLint`                    | `7.8.1` |
| `node`                      | `10.17.0` |

**Comment by EvHaus** [comment]
Looks like this actually affects `@typescript-eslint/eslint-plugin` and not `@typescript-eslint/parser`. My apologies.

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

First answer in the FAQ.
you're using the wrong rule.

Also this is a duplicate of #2502 - please use the search next time.
